src/sst/elements/memHierarchy/cacheController.cc

```diff
@@ -62,6 +62,13 @@ void Cache::handleEvent(SST::Event * ev) {
  *  a prefetch target
  */
 void Cache::handlePrefetchEvent(SST::Event * ev) {
+    MemEventBase *event = static_cast<MemEventBase*>(ev);
+
+    // Record the time at which requests arrive for latency statistics
+    if (CommandClassArr[(int)event->getCmd()] == CommandClass::Request &&
```

Comment:
Can you move this to the processPrefetchEvent funciton where the rest of the statistic handling is done? That will start the latency timer at the point at which the prefetch is recorded as 'received'.

Reply:
I updated the code as you commented.